src/data/camera.ts

```diff
@@ -26,6 +27,7 @@ export interface CameraEntity extends HassEntityBase {
 
 export interface CameraPreferences {
   preload_stream: boolean;
+  orientation: number;
```

Comment:
```ts
type ValueOf<T extends any[]> = T[number];

orientation: ValueOf<typeof CAMERA_ORIENTATIONS>;
```

Reply:
here is a nice article: https://dev.to/smeijer/a-typescript-valueof-implementation-and-how-it-s-built-4gim